htdocs/compta/facture/card.php

```diff
@@ -4801,6 +5173,30 @@ function js_recalculate_revenuestamp(){
 					print '<div class="inline-block divButAction"><a class="butAction" href="' . $_SERVER['PHP_SELF'] . '?socid=' . $object->socid .'&amp;fac_avoir=' . $object->id . '&amp;action=create&amp;type=2'.($object->fk_project > 0 ? '&amp;projectid='.$object->fk_project : '').'">' . $langs->trans("CreateCreditNote") . '</a></div>';
 				}
 			}
+			
+			// Fast create of credit note
+			if (($object->type == Facture::TYPE_STANDARD || $object->type == Facture::TYPE_DEPOSIT) && $object->statut > 0 && $user->rights->facture->creer)
+			{
+				if (! $objectidnext)
+				{
+					$qualified_for_stock_change = 0;
+					if (empty($conf->global->STOCK_SUPPORTS_SERVICES)) {
+						$qualified_for_stock_change = $object->hasProductsOrServices(2);
+					} else {
+						$qualified_for_stock_change = $object->hasProductsOrServices(1);
+					}
+					
+					if (! empty($conf->global->STOCK_CALCULATE_ON_BILL) && $qualified_for_stock_change)
+					{
+						print '<div class="inline-block divButAction"><a class="butAction" href="' . $_SERVER['PHP_SELF'] . '?facid=' . $object->id . '&amp;action=createCreditNoteFast&amp;type=2">' . $langs->trans("CreateCreditNoteFast") . '</a></div>';
```

Comment:
The purpose of this function is to avoid going through multiple screens and save time

Create credit note and charge the discount directly to the invoice.